src/monitor/node_active_protocol.c

```diff
@@ -965,7 +989,153 @@ stop_maintenance(PG_FUNCTION_ARGS)
+	text *nodeNameText = PG_GETARG_TEXT_P(1);
+	char *nodeName = text_to_cstring(nodeNameText);
+	int32 nodePort = PG_GETARG_INT32(2);
+	int candidatePriority = PG_GETARG_INT32(3);
+	AutoFailoverNode *currentNode = NULL;
+	bool nodeUpdated = false;
+
+	ereport(LOG, (errmsg("updating candidate_priority for node %d, candidate_priority=%d",
+						  nodeId, candidatePriority)));
+
+	checkPgAutoFailoverVersion();
+
+	currentNode = GetAutoFailoverNodeWithId(nodeId, nodeName, nodePort);
+
+	if (currentNode == NULL)
+	{
+		ereport(ERROR, (errmsg("node %d is not registered", nodeId)));
+	}
+
+	LockFormation(currentNode->formationId, ShareLock);
+	LockNodeGroup(currentNode->formationId, currentNode->groupId, ExclusiveLock);
+
+	if (candidatePriority < 0 || candidatePriority > 100)
+	{
+		ereport(ERROR, (ERRCODE_INVALID_PARAMETER_VALUE,
+						errmsg("invalid value for candidate_priority \"%d\" "
+							   "expected an integer value between 0 and 100", candidatePriority)));
+	}
+
+
+	if (candidatePriority != currentNode->candidatePriority)
+	{
+		currentNode->candidatePriority = candidatePriority;
+		nodeUpdated = true;
+	}
+
+	if (nodeUpdated)
+	{
+		char message[BUFSIZE];
+
+		ReportAutoFailoverNodeReplicationSetting(currentNode->nodeId,
+				currentNode->nodeName,
+				currentNode->nodePort,
+				currentNode->candidatePriority,
+				currentNode->replicationQuorum);
+
+		LogAndNotifyMessage(
+			message, BUFSIZE,
+			"Updating candidatePriority.");
+
+		NotifyStateChange(currentNode->reportedState,
+						  currentNode->goalState,
+						  currentNode->formationId,
+						  currentNode->groupId,
+						  currentNode->nodeId,
+						  currentNode->nodeName,
+						  currentNode->nodePort,
+						  currentNode->pgsrSyncState,
+						  currentNode->reportedLSN,
+						  currentNode->candidatePriority,
+						  currentNode->replicationQuorum,
+						  message);
+
+	}
+
+	PG_RETURN_BOOL(nodeUpdated);
+}
+
+
+/*
+ * set_node_replication_quorum sets node replication quorum property
+ */
+Datum
+set_node_replication_quorum(PG_FUNCTION_ARGS)
+{
+	int32 nodeid = PG_GETARG_INT32(0);
+	text *nodeNameText = PG_GETARG_TEXT_P(1);
+	char *nodeName = text_to_cstring(nodeNameText);
+	int32 nodePort = PG_GETARG_INT32(2);
+	bool replicationQuorum = PG_GETARG_BOOL(3);
+	AutoFailoverNode *currentNode = NULL;
+	bool nodeUpdated = false;
+
+	ereport(LOG, (errmsg("updating node replication_quorum for node %d, replication_quorum=%s",
+						  nodeid, replicationQuorum?"true":"false")));
+
+	checkPgAutoFailoverVersion();
+
+	currentNode = GetAutoFailoverNodeWithId(nodeid, nodeName, nodePort);
+
+	if (currentNode == NULL)
+	{
+		ereport(ERROR, (errmsg("node %d is not registered", nodeid)));
+	}
+
+	LockFormation(currentNode->formationId, ShareLock);
+	LockNodeGroup(currentNode->formationId, currentNode->groupId, ExclusiveLock);
+
+	if (replicationQuorum != currentNode->replicationQuorum)
+	{
+		currentNode->replicationQuorum = replicationQuorum;
+		nodeUpdated = true;
+	}
```

Comment:
Same comment as before about the semantics (returning false when we are setting the same value again) and race-conditions (get before lock, then use a value that might have been updated already). Again my preference is to not do that test at all and let Postgres handle the concurrency here as far as new values are concerned (let's still lock the formation).